digits.go

```diff
@@ -21,41 +22,43 @@ import (
 type digits [10]rune
 
 func (d digits) twoDigit(number int) string {
-	if number < 10 { //nolint:mnd
-		return string(d[0]) + string(d[number])
-	}
+	ones := number % 10      //nolint:mnd
+	tens := number / 10 % 10 //nolint:mnd
 
-	last := number % 10 //nolint:mnd
-	number /= 10
-	beforeLast := number % 10 //nolint:mnd
+	// latn
+	if d[0] == '0' {
```

Comment:
The numbering system is known when the DateTimeFormat is initialised. I don't have to check it repeatedly when I format all numbers.